### PR TITLE
[pg_sni_router] Add session_id to more messages

### PR DESCRIPTION
**Pull request by ololobus**
## Problem

Currently, we only have `session_id` in the `handle_client` span, so it's really hard to track the whole connection path in the logs. Especially if error occurs.

Related to neondatabase/cloud#5265

## Summary of changes

This commit adds `session_id` explicitly into several other messages. This could be a bit too verbose, but I've tried several combinations of `span.in_scope()` and `instrument` and it didn't play well, because all related messages are in the different contexts.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: proxy/src/bin/pg_sni_router.rs [hunk 1]
```diff
@@ -141,25 +141,35 @@ async fn task_main(
...
+                        );
 
                         socket
                             .set_nodelay(true)
                             .context("failed to set socket option")?;
 
                         handle_client(dest_suffix, tls_config, session_id, socket).await
                     }
-                    .unwrap_or_else(|e| {
+                    .unwrap_or_else(move |e| {
```
Comment (ololobus) [inline proxy/src/bin/pg_sni_router.rs:167]: Not sure about this move, it's needed to use session_id bellow

Comment (koivunej) [reply proxy/src/bin/pg_sni_router.rs:167]: Did compiler want it? I don't see why it would be needed here.

Comment (koivunej) [inline proxy/src/bin/pg_sni_router.rs:167]: Instead I think we should add a to capture everything this call to the session.

The `unwrap_or_else` is on a strange position, but not changed here, I don't think I understand it's function, need to look at this in an editor :)

Comment (koivunej) [reply proxy/src/bin/pg_sni_router.rs:167]: Aah, silly me, it's a `FutureExt::unwrap_or_else`.

Comment (ololobus) [reply proxy/src/bin/pg_sni_router.rs:167]: Yes, it was complaining about it without move

**Comment by ololobus** [comment]
Closing as @koivunej proposed a better solution

**Pull request closed without merging**